cli/flox/doc-catalog/manifest.toml.md

```diff
@@ -0,0 +1,368 @@
+profile scripts, commands, and shells.
+
+Hook scripts inherit environment variables set in the `[vars]` section,
+and variables set here will in turn be inherited by the `[profile]` scripts
+described below.
+
+Any output written to `stdout` in a hook script is redirected to `stderr` to
+avoid it being mixed with the output of profile section scripts that write to
+`stdout` for "in-place" activations.
+
+```toml
+[hook]
+on-activate = """
+    # Interact with the tty as you would in any script
+    echo "Starting up $FLOX_ENV_DESCRIPTION environment ..."
+    read -e -p "Favourite colour or favorite color? " value
+
+    # Set variables, create files and directories
+    venv_dir="$(mktemp -d)"
+    export venv_dir
+
+    # Perform initialization steps, e.g. create a python venv
+    python -m venv "$venv_dir"
+
+    # Invoke apps that configure the environment via stdout
+    eval "$(ssh-agent)"
+"""
+```
+
+The `on-activate` script is not re-run when activations are nested.
+A nested activation can occur when an environment is already active and either
+`eval "$(flox activate)"` or `flox activate -- CMD` is run.
+In this scenario, `on-activate` is not re-run.
+Currently, environment variables set by the first run of the `on-activate`
+script are captured and then later set by the nested activation,
+but this behavior may change.
+
+It is best to write hooks defensively, assuming the user is using the
+environment from any directory on their machine.
+
+### `script` - DEPRECATED
+This field was deprecated in favor of the `profile` section.
+It will be removed in a later release.
+
+## `[profile]`
+
+Scripts defined in the `[profile]` section are sourced by *your shell* and
+inherit environment variables set in the `[vars]` section and by the `[hook]`
+scripts.
+The `profile.common` script is sourced for every shell,
+and special care should be taken to ensure compatibility with all shells.
+The `profile.<shell>` scripts are then sourced *after* `profile.common` by the
+corresponding shell.
+
+These scripts are useful for performing shell-specific customizations such as
+setting aliases or configuring the prompt.
+
+```toml
+[profile]
+common = """
+    echo "it's gettin' flox in here"
+"""
+bash = """
+    source $venv_dir/bin/activate
+    alias foo="echo bar"
+    set -o vi
+"""
+zsh = """
+    source $venv_dir/bin/activate
+    alias foo="echo bar"
+    bindkey -v
+"""
+```
+
+Profile scripts are re-run for nested activations.
+A nested activation can occur when an environment is already active and either
+`eval "$(flox activate)"` or `flox activate -- CMD` is run.
+In this scenario, profile scripts are run a second time.
+Re-running profile scripts allows aliases to be set in subshells that inherit
+from a parent shell with an already active environment.
+
+## `[options]`
+
+The `[options]` section of the manifest details settings for the environment
+itself.
+
+The most common option to set is `systems`,
```

Comment:
this will no longer be mostly common for users once the catalog service is out as we will render/eval the popular system types automatically. I would expect if someone installs something that fails for a specific system their corrective action should be nudged towards making the package system-specific verses the environment